pkg/machinery/meta/meta.go

```diff
@@ -66,6 +90,23 @@ func DecodeValues(s string) (Values, error) {
 		return nil, nil
 	}
 
+	// do un-gzip if needed
+	if hasGzipMagic(b) {
+		gzR, err := gzip.NewReader(bytes.NewReader(b))
+		if err != nil {
+			return nil, err
+		}
+
+		b, err = io.ReadAll(gzR)
+		if err != nil {
+			return nil, err
+		}
+
+		if err := gzR.Close(); err != nil {
+			return nil, err
+		}
```

Comment:
shouldn't we do this using `defer` - to make sure it closes unconditionally even if `ReadAll` fails?

Reply:
I think it'd be more clear, but I think it's not relevant, `.Close()` verifies the checksum, it actually doesn't close (like deallocate) anything

Reply:
updated with deferred Close